script.js

```diff
@@ -1,12 +1,239 @@
+  }
+});
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  let apiUrl = `https://api.tvmaze.com/shows/527/episodes`;
+  getDataFromAPI(apiUrl);
+
+  //level 400
+  allShows = getAllShows().sort((a, b) => {
+    const nameA = a.name.toUpperCase(); // ignore upper and lowercase
+    const nameB = b.name.toUpperCase(); // ignore upper and lowercase
+    if (nameA < nameB) {
+      return -1;
+    }
+    if (nameA > nameB) {
+      return 1;
+    }
+
+    // names must be equal
+    return 0;
+  });
+
+  allShows.forEach((el, inx) => {
+    let showsSelectOptEl = document.createElement("option");
+    showsSelectOptEl.text = allShows[inx].name;
+    showsSelectOptEl.value = inx;
+    showsSelectEl.add(showsSelectOptEl);
+  });
+}
+
+window.onload = setup;
+
+function getDataFromAPI(myAPI) {
+  fetch(myAPI)
+    .then((response) => {
+      if (response.status >= 200 && response.status <= 299) {
+        console.log("success", response);
+        return response.json();
+      } else {
+        throw new Error(
+          `Encountered something unexpected: ${response.status} ${response.statusText}`
+        );
+      }
+    })
+    .then((jsonResponse) => {
+      // do whatever you want with the JSON response
+      let allEpisodes = jsonResponse;
+      makePageForEpisodes(allEpisodes);
+    })
+    .catch((error) => {
+      // Handle the error
+      console.log(error);
+    });
 }
 
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  let results = episodeList.length;
+  document.querySelector(
+    ".searchDescEl"
+  ).innerHTML = `   Displaying  <span id="myResults">${results}</span> of ${results} Results`;
+
+  //level 200
+  // Add a "live" search input:
+  // Only episodes whose summary OR name contains the search term should be displayed
+  searchInputEl.addEventListener("input", () => {
+    let filterList = episodeList.filter((el) => {
+      let inputChar = searchInputEl.value;
+      return (
+        //case insensitive
+        el.name.toUpperCase().includes(inputChar.toUpperCase()) ||
+        el.summary.toUpperCase().includes(inputChar.toUpperCase())
+      );
+    });
+
+    if (filterList.length) {
+      showEpisodes(filterList);
+    } else {
+      while (showsEl.firstChild) {
+        showsEl.removeChild(showsEl.firstChild);
+      }
+      document.getElementById("myResults");
+      myResults.textContent = "0";
+    }
+  });
+
+  //initial showing of episodes - refactored in level 200 to make easier for reusability
+  showEpisodes(episodeList);
+
+  //event listener for select element
+  searchSelectEl.addEventListener("change", function () {
+    if (this.value === "Click for All Episodes") {
+      showEpisodes(episodeList);
+    } else {
+      let singleArr = [];
+      singleArr.push(episodeList[this.value]);
+      showEpisodes(singleArr);
+    }
```

Comment:
it looks like theres always going to be 1 value in that array? so having the code to initialise and add to the array isnt required.

Reply:
I didnt know how to have this in array form. perhaps we can work through these in our next buddy group.